plugins/ingest/src/main/java/org/elasticsearch/ingest/processor/mutate/MutateProcessor.java

```diff
@@ -0,0 +1,312 @@
+                newVal = parseValueAsType(oldVal, toType);
+            }
+
+            data.addField(entry.getKey(), newVal);
+        }
+    }
+
+    private void doSplit(Data data) {
+        if (split == null) { return; }
+
+        for(Map.Entry<String, String> entry : split.entrySet()) {
+            Object oldVal = data.getProperty(entry.getKey());
+            if (oldVal instanceof String) {
+                data.addField(entry.getKey(), Arrays.asList(((String) oldVal).split(entry.getValue())));
+            } else {
+                throw new IllegalArgumentException("Cannot split a field that is not a String type");
+            }
+        }
+    }
+
+    private void doGsub(Data data) {
+        if (gsub == null) { return; }
+
+        for (Map.Entry<String, List<String>> entry : gsub.entrySet()) {
+            String fieldName = entry.getKey();
+            List<String> matchAndReplaceList = entry.getValue();
+            String oldVal = data.getProperty(fieldName);
+            String newVal = oldVal.replaceAll(matchAndReplaceList.get(0), matchAndReplaceList.get(1));
+            data.addField(entry.getKey(), newVal);
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    private void doJoin(Data data) {
+        if (join == null) { return; }
+
+        for(Map.Entry<String, String> entry : join.entrySet()) {
+            Object oldVal = data.getProperty(entry.getKey());
+            if (oldVal instanceof List) {
+                String joined = (String) ((List) oldVal)
+                        .stream()
+                        .map(Object::toString)
+                        .collect(Collectors.joining(entry.getValue()));
+
+                data.addField(entry.getKey(), joined);
+            } else {
+                throw new IllegalArgumentException("Cannot join field:" + entry.getKey() + " with type: " + oldVal.getClass());
+            }
+        }
+    }
+
+    private void doRemove(Data data) {
+        if (remove == null) { return; }
+        for(String field : remove) {
+            data.getDocument().remove(field);
+        }
+    }
+
+    private void doStrip(Data data) {
+        if (strip == null) { return; }
+
+        for(String field : strip) {
+            Object val = data.getProperty(field);
+            if (val instanceof String) {
+                data.addField(field, ((String) val).trim());
+            } else {
+                throw new IllegalArgumentException("Cannot strip field:" + field + " with type: " + val.getClass());
+            }
+        }
+    }
+
+    private void doUppercase(Data data) {
+        if (uppercase == null) { return; }
+
+        for(String field : uppercase) {
+            Object val = data.getProperty(field);
+            if (val instanceof String) {
+                data.addField(field, ((String) val).toUpperCase(Locale.US));
+            } else {
+                throw new IllegalArgumentException("Cannot uppercase field:" + field + " with type: " + val.getClass());
```

Comment:
maybe this can be decided by the user using something like the `continue_on_failure` flag.